src/snowcli/cli/common/flags.py

```diff
@@ -1,28 +1,35 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import Any, Callable, Optional
 
 import typer
-from snowcli.cli.common.snow_cli_global_context import (
-    ConnectionDetails,
-    update_callback,
-)
+from snowcli.cli.common.cli_global_context import cli_context_manager
 from snowcli.output.formats import OutputFormat
 
 DEFAULT_CONTEXT_SETTINGS = {"help_option_names": ["--help", "-h"]}
 
-
 _CONNECTION_SECTION = "Connection configuration"
 _CLI_BEHAVIOUR = "Global configuration"
 
 
+def _callback(provide_setter: Callable[[], Callable[[Any], Any]]):
```

Comment:
```suggestion
def _callback(setter: Callable[[], Callable[[Any], Any]]):
```
Won't it be enough?

Reply:
IMO `provide_setter` sounds like a function which returns a  function, while just `setter` sounds like a function which is a setter it self